### PR TITLE
Feature | bulk upload handle pre validate missing mandatory file

### PR DESCRIPTION
**Pull request by omarCreativeDev**
- handle pre validate missing mandatory file
- created BulkUploadFileType enum, utilize enum in template
- on files selection set filesUploaded to false so that user can re choose files
- cleanup `onValidateComplete()` method

### CODE REVIEW

File: src/app/features/bulk-upload/files-upload/files-upload.component.ts [hunk 1]
```diff
@@ -25,7 +21,7 @@ export class FilesUploadComponent implements OnInit {
   private selectedFiles: File[];
   private bytesTotal = 0;
   private bytesUploaded: number[] = [];
-  private presignedUrlsSubcription$: Subscription = new Subscription();
+  private subcriptions$: Subscription = new Subscription();
```
Comment (wozzer72) [inline src/app/features/bulk-upload/files-upload/files-upload.component.ts]: I can see the uploaded files have their own subscription set and aptly named `uploadSubscription`. What is this set of subscriptions for and wouldn't it be prudent to explicitly name the context for these `subscriptions` - e.g. `prevalidateSubscriptions`?

File: src/app/features/bulk-upload/files-upload/files-upload.component.ts [hunk 2]
```diff
@@ -152,7 +163,7 @@ export class FilesUploadComponent implements OnInit {
 
   public cancelUpload(): void {
     this.filesUploading = false;
-    this.presignedUrlsSubcription$.unsubscribe();
+    this.subcriptions$.unsubscribe();
```
Comment (wozzer72) [inline src/app/features/bulk-upload/files-upload/files-upload.component.ts]: Refer also to the generic naming of this subscription set above.

File: src/app/features/bulk-upload/uploaded-files-list/uploaded-files-list.component.html [hunk 3]
```diff
@@ -39,13 +39,13 @@
           </p>
         </td>
         <td class="govuk-table__cell" [class.govuk-error-table__cell]="file.errors">
-          {{ file.fileType ? file.fileType : getFileType(file.filename) }}
+          {{ file.fileType ? bulkUploadFileTypeEnum[file.fileType] : getFileType(file.filename) }}
```
Comment (wozzer72) [inline src/app/features/bulk-upload/uploaded-files-list/uploaded-files-list.component.html]: I presume this is mapping old names, like establishment and worker to workplace and staff?

File: src/app/features/bulk-upload/files-upload/files-upload.component.ts (hunk 1 above)
Comment (omarCreativeDev) [reply src/app/features/bulk-upload/files-upload/files-upload.component.ts]: thanks, have changed

File: src/app/features/bulk-upload/files-upload/files-upload.component.ts (hunk 2 above)
Comment (omarCreativeDev) [reply src/app/features/bulk-upload/files-upload/files-upload.component.ts]: done

File: src/app/features/bulk-upload/uploaded-files-list/uploaded-files-list.component.html (hunk 3 above)
Comment (omarCreativeDev) [reply src/app/features/bulk-upload/uploaded-files-list/uploaded-files-list.component.html]: yup thats right, FE just displaying the copy as per design

**Comment by omarCreativeDev** [comment]
[Bulk Upload - Pre Validate Files upon Upload](https://trello.com/c/Rrn2NBMb/110-bulk-upload-pre-validate-files-upon-upload)

**Pull request merged**